src/Kernel32.Desktop/Kernel32.Helpers.cs

```diff
@@ -197,26 +197,27 @@ public static IEnumerable<MODULEENTRY32> Module32Enumerate(SafeObjectHandle hSna
             }
         }
 
-        public static string QueryFullProcessImageName(
+        public static unsafe string QueryFullProcessImageName(
             SafeObjectHandle hProcess,
             QueryFullProcessImageNameFlags dwFlags = QueryFullProcessImageNameFlags.None)
         {
             // If we ever resize over this value something got really wrong
             const int maximumRealisticSize = 1 * 1024 * 2014;
 
-            var buffer = new StringBuilder(255);
+            int size = 255;
+
             do
             {
-                int size = buffer.Capacity;
-                bool success = QueryFullProcessImageName(hProcess, dwFlags, buffer, ref size);
+                var text = stackalloc char[size];
```

Comment:
> This is very interesting information that you brought us, I don't follow your calculation of 3.5 Mo

I don't follow my computations either as I find 3.98 Mo now, anyway I didn't really compute, I just ran the code :

```csharp
const int maximumRealisticSize = 1 * 1024 * 2014;
int size = 255;
long accum = 0;
do
{
	accum += sizeof(char) * size;
	size *= 2;
}
while (size < maximumRealisticSize);

// Write 3.98388862609863
Console.WriteLine(accum / 1024.0 / 1024.0);
```

(Windows unicode chars are UTF16, so 2 bytes, the `sizeof(char)` is important)

> Anyway, may I use a simple new char[size] here? Isn't the managed array be incompatible with the unsafe pointer? 

Yes it's incompatible but you just have to pin it :

```csharp
	do
	{
		fixed (char* text = new char[size])
		{
			var success = QueryFullProcessImageName(hProcess, dwFlags, text, ref size);
			if (success)
			{
				return new string(text, 0, size);
			}
		}

		size *= 2;
	}
	while (size < maximumRealisticSize);
```

 Alternatively you can go full low level, it's better perf-wise (Doesn't bother the GC just to pin the memory immediately and free it 1ms later) but I don't think it's a good idea here as it make the code a litter harder to read  :

````csharp
	do
	{
		char* text = (char*)Marshal.AllocHGlobal(size * sizeof(char));
		try
		{
			var success = QueryFullProcessImageName(hProcess, dwFlags, text, ref size);
			if (success)
			{
				return new string(text, 0, size);
			}
		}
		finally
		{
			Marshal.FreeHGlobal((IntPtr)text);
		}

		size *= 2;
	}
	while (size < maximumRealisticSize);
```